kitsune/forums/jinja2/forums/includes/post.html

```diff
@@ -30,26 +31,31 @@
       {% endif %}
       {% if post.id and not thread.is_locked and
         has_perm_or_owns('forums_forum.post_edit_forum', post, forum, 'author') %}
-        <li class="mzp-c-menu-list-item"><a href="{{ url('forums.edit_post', forum.slug, thread.id, post.id) }}">{{ _('Edit this post') }}</a></li>
+      <li class="mzp-c-menu-list-item"><a
+          href="{{ url('forums.edit_post', forum.slug, thread.id, post.id) }}">{{ _('Edit this post') }}</a></li>
       {% endif %}
       {% if post.id and has_perm('forums_forum.post_delete_forum', forum) %}
-        <li class="mzp-c-menu-list-item"><a href="{{ url('forums.delete_post', forum.slug, thread.id, post.id) }}">{{ _('Delete this post') }}</a></li>
+      <li class="mzp-c-menu-list-item"><a
+          href="{{ url('forums.delete_post', forum.slug, thread.id, post.id) }}">{{ _('Delete this post') }}</a></li>
       {% endif %}
 
-      {% if post.id and user.is_authenticated() and not thread.is_locked and forum.allows_posting_by(user) %}
+      {% if user.is_authenticated() %}
+      {% if post.id and not thread.is_locked and forum.allows_posting_by(user) %}
       <li class="mzp-c-menu-list-item post-action">
         <a class="reply" data-post="{{ post.id }}" href="#thread-reply">{{ _('Quote') }}</a>
       </li>
-    {% endif %}
+      {% endif %}
+
+      {% if post.id %}
```

Comment:
Should we move these common `if post.id` conditions into the `if user.is_authenticated()` conditional?